d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -280,6 +291,18 @@ private void handleResourceUpdate(Map<String, ? extends ResourceUpdate> updates,
     }
   }
 
+  private void handleResourceRemoval(List<String> removedResources, ResourceType type)
+  {
```

Comment:
we could add a "resourceNotFoundCount" metric to XdsClientJmx, and increment it by the size of the removedResources list here.